RFC-0039-generalize-pytorch-ut.md

```diff
@@ -0,0 +1,52 @@
+
+# [RFC] Generalization of PyTorch framework UT for non-cuda device execution
+
+**Authors:**
+* @ankurneog
+
+
+## **Summary**
+Modify PyTorch framework UTs so that non-cuda devices such as intel Gaudi and intel XPU is able to harness the content and improve quality.
+
+
+## **Motivation**
+The Pytorch framework UTs are good indicator for device stack health, however these are mostly written for cpu and cuda devices, which restricts its use for non-cuda devices.
+
+We propose to modify the content wherever possible to make it available for non-cuda device execution
+
+This will also ensure greater participation for content enhancement.
+
+## **Proposed Implementation**
```

Comment:
I think the idea makes a lot of sense.
I think we would need more details and how this interracts with existing features like the device-generic tests (https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/common_device_type.py) that already work for privateuse1 partially btw and the opinfo consistency tests.

Also I don't think we want to aim at running the full test suite with the side device available but select specific device-dependent tests that need to be ran for each device we support.

Reply:
@albanD  : thanks for your comment , yes I believe we need some extensive hooks to enable this , for eg. the one we introduced with 
https://github.com/pytorch/pytorch/pull/128584/files#diff-d183f2afc51d6a59bc70094e8f476d2468c45e415500f6eb60abad955e065156R531
```@onlyNativeDeviceTypesAnd(["hpu"])```
The other devices can add to to such list, if it supports the TC.

we can modify other hooks like skipIfDevice in similar fashion.

The common_device_type is useable if we replace the ```onlyNativeDeviceType```  decorator. 
It was widely used in the initial files but all recent files are not using it (eg: dynamo/distributed) and instead directly make ```.cuda()``` calls. 

However these content shouldn't be too difficult to migrate. 

I believe in general we should ensure  new test content uses the common_device_type framework, and open up the content for "non-native" device execution.